idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/application/cache/PersistentService.kt

```diff
@@ -0,0 +1,43 @@
+package arrow.meta.ide.dsl.application.cache
+
+import com.intellij.openapi.components.PersistentStateComponent
+
+/**
+ * [PersistentService] caches a [value] and solely allows monomorphic updates.
+ * This Service is one way to replace global variables stored in components to their service representation,
+ * store state in either project-level or application-level, as an effort to migrate to Services.
+ * This interface shouldn't be used to store PsiElements or VirtualFiles.
+ */
+interface PersistentService<A> : PersistentStateComponent<Id<A>> {
```

Comment:
Review notes of https://github.com/arrow-kt/arrow-meta/pull/490#pullrequestreview-381077404 apply here